lib/dal/src/pkg/export.rs

```diff
@@ -810,9 +810,18 @@ impl PkgExporter {
             AttributePrototypeArgument::list_ids_for_prototype(ctx, prototype_id).await?;
 
         // If the prototype func is intrinsic and has no arguments, it's one that is created by default
-        // and we don't have to track it in the package
-        if apas.is_empty() && is_intrinsic(proto_func.name.as_str()) {
-            return Ok(None);
+        // and we don't have to track it in the package. However, if it was formerly a special case
+        // func, then we will continue to treat it as such.
+        if apas.is_empty() {
+            match IntrinsicFunc::maybe_from_str(proto_func.name.as_str()) {
+                Some(IntrinsicFunc::ResourcePayloadToValue)
+                | Some(IntrinsicFunc::NormalizeToArray)
+                    if proto_func.backend_kind == FuncBackendKind::JsAttribute => {}
+                Some(_) => {
+                    return Ok(None);
+                }
+                None => {}
+            }
```

Comment:
You'll see this `IntrinsicFunc::maybe_from_str` pattern a lot where we check the func kind or backend kind. That is us accounting for the old funcs. That contributed a huge part to why this PR stayed in drafts  for a chunk of time: lot of papercuts.